tfx/components/infra_validator/model_server_runners/kubernetes_runner.py

```diff
@@ -208,6 +212,9 @@ def _BuildPodManifest(self) -> k8s_client.V1Pod:
     if isinstance(self._serving_binary, serving_bins.TensorFlowServing):
       env_vars_dict = self._serving_binary.MakeEnvVars(
           model_path=self._model_path)
+      for key, value in self._custom_env_vars_dict.items():
```

Comment:
nit: how about simply

```python
if self._config.serving_pod_spec_overrides:
  env_vars_dict.update(
      self._config_serving_pod_spec_overrides.environment_variables)
```

Reply:
The reason for the current logic was that I didn't want the overrides to override the IV specific env vars such as tf servings MODEL.

I'm happy to change to your proposal if you think it's fine.